packages/core/src/processed-data-manager.ts

```diff
@@ -0,0 +1,134 @@
+import get from 'lodash/get';
+import { ApplicationError } from 'n8n-workflow';
+import type {
+	IProcessedDataManager,
+	ICheckProcessedOptions,
+	ICheckProcessedOutput,
+	ICheckProcessedOutputItems,
+	IDataObject,
+	ProcessedDataContext,
+	ProcessedDataItemTypes,
+	ICheckProcessedContextData,
+} from 'n8n-workflow';
+
+export class ProcessedDataManager {
+	private static instance: ProcessedDataManager;
+
+	private manager: IProcessedDataManager;
```

Comment:
This is a bit confusing as the class is a `ProcessedDataManager`, but then it has a `manager` field which is a `IProcessedDataManager` interface. Could somehow clarify these different parts? If I understood this correctly, this implements a singleton pattern and delegates almost everything to the instance. Could we somehow clarify what these different parts are? E.g. name this `CoreProcessedDataManager` or `ProcessedDataManagerFacade` or something like that?

Reply:
it is actually a singleton, I renamed the class and the property hopefully it's clearer now